server/src/main/java/org/opensearch/cluster/action/shard/ShardStateAction.java

```diff
@@ -222,6 +234,52 @@ public void handleException(TransportException exp) {
         }
     }
 
+    /**
+     * RetryableAction for performing retires for cluster manager throttling.
+     */
+    private class ShardStateClusterManagerAction extends RetryableAction {
```

Comment:
Can we come up with class and move all the logic to a common class. Looks redundant

Reply:
Replied in above comment.